eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1701,16 +1701,18 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     // a channel_reestablish when reconnecting a channel that recently got confirmed, and instead send a funding_locked
     // first and then go silent. This is due to a race condition on their side, so we trigger a reconnection, hoping that
     // we will eventually receive their channel_reestablish.
-    case Event(_: FundingLocked, _) =>
+    case Event(_: FundingLocked, d) =>
       log.warning("received funding_locked before channel_reestablish (known lnd bug): disconnecting...")
+      send(Warning(d.channelId, "spec violation: you sent funding_locked before channel_reestablish"))
       peer ! Peer.Disconnect(remoteNodeId)
       stay
```

Comment:
I'm guessing we are going to send a few warnings 😜 

The preferred way would be:
```suggestion
      val warning = Warning(d.channelId, "spec violation: you sent funding_locked before channel_reestablish")
      peer ! Peer.Disconnect(remoteNodeId)
      stay sending warning
```

The goal is to have side effects occur outside the handlers themselves.

Reply:
But then we'll disconnect before sending the warning, won't we?

Reply:
True. I believe a comment is in order!

There is always a risk though, because sending is asynchronous, but it was much less likely before.

Reply:
I added a delay before the `Peer.Disconnect` in https://github.com/ACINQ/eclair/pull/1863/commits/bf10f462f50cd45c865e59370b572a42f736cd39 this way we can send the warning outside of the handler with `sending`.